### ISSUE TITLE
Could not find init script or upstart conf file for 'sensu-enterprise'

### ISSUE THREAD

**Issue by atrepca** [sensu/sensu-puppet#662]
## Description of problem

- What did you do?
We upgraded the Sensu Puppet Forge module from version 2.1.0 to version 2.2.1, and the Sensu version from 0.25.7-1 to 0.26.5-2.
- What happened?
Puppet is failing due to not finding the init script/ upstart conf file for the `sensu-enterprise` service.
- What did you expect to happen?
We're using Sensu Core not Sensu Enterprise; we don't have any Enterprise packages installed, so that condition will never be met. There shouldn't be any overlap.
- How can someone reproduce the problem?

## Command used and debugging output

- What does your puppet config look like (including any hiera config)
```
  class { '::sensu':
    version                  => '0.26.5-2',
    rabbitmq_ssl_private_key => '/etc/sensu/ssl/key.pem',
    rabbitmq_ssl_cert_chain  => '/etc/sensu/ssl/cert.pem',
    rabbitmq_password        => $rabbitmq_password,
    rabbitmq_port            => '5671',
    rabbitmq_host            => $rbmq_host,
    rabbitmq_vhost           => 'sensu',
    subscriptions            => 'sensu-prod',
    purge                    => true,
    safe_mode                => true,
    use_embedded_ruby        => true,
    sensu_plugin_version     => present,
    client_custom            => {
      'team'       => "${::app}-${::env}",
      'search_tag' => $search_tag
    },
    client_keepalive         => {
      'handlers'   => $keepalive_handlers,
      'refresh'    => 14400,
      'thresholds' => {
        'warning'  => 120,
        'critical' => 180,
      },
    },
    repo_source              => 'http://repositories.sensuapp.org/apt',
    repo_key_id              => 'EE15CFF6AB6E4E290FDAB681A20F259AEB9C94BB',
    repo_key_source          => 'http://repositories.sensuapp.org/apt/pubkey.gpg',
  }
```
- Is this a masterless or master based puppet setup?
Master based.

## Platform and version information

- Your OS: Ubuntu 14.04.1 LTS
- Your Ruby version: 1.9.3p484
- Your version of Puppet: 3.7.2, open source
- Your version of Sensu: 0.26.5-2
- Your version of this module: 2.2.1

## Anything else to add that you think will be helpful?
The [`default` case](https://github.com/sensu/sensu-puppet/blob/c7cdeb6a1ef5bc0b83f8c4c7e683e844a3dd5745/manifests/enterprise/service.pp#L28-L31) in the `sensu::enterprise::service` class needs an init script or upstart config file to ensure the `stopped` state for the service.

**Comment by atrepca** [comment]
Just saw this was fixed in #409 but then rolled back in #656. What was the reason for flipping the vars @dzeleski?

**Comment by dzeleski** [comment]
409 broke the ability to set manage_services to false and manage the services outside of the sensu module for enterprise users. Is puppet attempting to stop/disable the sensu-enterprise service?

Could you post a debug output of your puppet run?

**Comment by atrepca** [comment]
Yup @dzeleski, Puppet is attempting to stop/disable the `sensu-enterprise` service. Below the relevant bits from a debug run:
```
Debug: Service[sensu-enterprise](provider=upstart): Could not find sensu-enterprise.conf in /etc/init
Debug: Service[sensu-enterprise](provider=upstart): Could not find sensu-enterprise.conf in /etc/init.d
Debug: Service[sensu-enterprise](provider=upstart): Could not find sensu-enterprise in /etc/init
Debug: Service[sensu-enterprise](provider=upstart): Could not find sensu-enterprise in /etc/init.d
Debug: Service[sensu-enterprise](provider=upstart): Could not find sensu-enterprise.sh in /etc/init
Debug: Service[sensu-enterprise](provider=upstart): Could not find sensu-enterprise.sh in /etc/init.d
Error: /Stage[main]/Sensu::Enterprise::Service/Service[sensu-enterprise]: Could not evaluate: Could not find init script or upstart conf file for 'sensu-enterprise'
Notice: /Stage[main]/Sensu::Enterprise::Dashboard/Anchor[sensu::enterprise::dashboard::begin]: Dependency Service[sensu-enterprise] has failures: true
Warning: /Stage[main]/Sensu::Enterprise::Dashboard/Anchor[sensu::enterprise::dashboard::begin]: Skipping because of failed dependencies
Notice: /Stage[main]/Sensu::Enterprise::Dashboard::Config/File[/etc/sensu/dashboard.json]: Dependency Service[sensu-enterprise] has failures: true
Warning: /Stage[main]/Sensu::Enterprise::Dashboard::Config/File[/etc/sensu/dashboard.json]: Skipping because of failed dependencies
Notice: /Stage[main]/Sensu::Enterprise::Dashboard::Config/Sensu_enterprise_dashboard_config[corp-tech-api-mdb0-stg.2u.com]: Dependency Service[sensu-enterprise] has failures: true
Warning: /Stage[main]/Sensu::Enterprise::Dashboard::Config/Sensu_enterprise_dashboard_config[corp-tech-api-mdb0-stg.2u.com]: Skipping because of failed dependencies
Notice: /Stage[main]/Sensu::Enterprise::Dashboard::Service/Service[sensu-enterprise-dashboard]: Dependency Service[sensu-enterprise] has failures: true
Warning: /Stage[main]/Sensu::Enterprise::Dashboard::Service/Service[sensu-enterprise-dashboard]: Skipping because of failed dependencies
Notice: /Stage[main]/Sensu::Enterprise::Dashboard/Anchor[sensu::enterprise::dashboard::end]: Dependency Service[sensu-enterprise] has failures: true
Warning: /Stage[main]/Sensu::Enterprise::Dashboard/Anchor[sensu::enterprise::dashboard::end]: Skipping because of failed dependencies
Notice: /Stage[main]/Sensu/Anchor[sensu::end]: Dependency Service[sensu-enterprise] has failures: true
Warning: /Stage[main]/Sensu/Anchor[sensu::end]: Skipping because of failed dependencies
```

**Comment by atrepca** [comment]
Fixed by #660, closing.
